cfme/ansible/credentials.py

```diff
@@ -0,0 +1,340 @@
+    credential_type = Text(locator=".//label[normalize-space(.)='Credential type']/../div")
+    save_button = Button("Save")
+    reset_button = Button("Reset")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_ansible_credentials and
+            self.title.text == 'Edit a Credential "{}"'.format(self.context["object"].name)
+        )
+
+    def before_fill(self, values):
+        for name in self.widget_names:
+            if name not in values or values[name] is None:
+                continue
+            widget = getattr(self, name)
+            title = self.browser.get_attribute("title", widget)
+            try:
+                self.input(title).enable
+            except NoSuchElementException:
+                continue
+
+
+class CredentialsCollection(Navigatable):
+    """Collection object for the :py:class:`Credential`."""
+
+    def create(self, name, credential_type, **credentials):
+        add_page = navigate_to(self, "Add")
+        machine_credential_fill_dict = {
+            "username": credentials.get("username"),
+            "password": credentials.get("password"),
+            "private_key": credentials.get("private_key"),
+            "private_key_phrase": credentials.get("private_key_phrase"),
+            "privilage_escalation": credentials.get("privilage_escalation"),
+            "privilage_escalation_username": credentials.get("privilage_escalation_username"),
+            "privilage_escalation_password": credentials.get("privilage_escalation_password"),
+            "vault_password": credentials.get("vault_password")
+        }
+        scm_credential_fill_dict = {
+            "username": credentials.get("username"),
+            "password": credentials.get("password"),
+            "private_key": credentials.get("private_key"),
+            "private_key_phrase": credentials.get("private_key_phrase")
+        }
+        amazon_credential_fill_dict = {
+            "access_key": credentials.get("access_key"),
+            "secret_key": credentials.get("secret_key"),
+            "sts_token": credentials.get("sts_token"),
+        }
+        vmware_credential_fill_dict = {
+            "username": credentials.get("username"),
+            "password": credentials.get("password"),
+            "vcenter_host": credentials.get("vcenter_host")
+        }
+        credential_type_map = {
+            "Machine": machine_credential_fill_dict,
+            "Scm": scm_credential_fill_dict,
+            "Amazon": amazon_credential_fill_dict,
+            "VMware": vmware_credential_fill_dict
+        }
+
+        add_page.fill({"name": name, "credential_type": credential_type})
+        add_page.credential_form.fill(credential_type_map[credential_type])
+        add_page.add_button.click()
+        credentials_list_page = self.create_view(CredentialsListView)
+        # Without this StaleElementReferenceException can be raised
+        wait_for(lambda: False, silent_failure=True, timeout=5)
```

Comment:
I encountered with sporadic `StaleElementReferenceException` without waiting a while after page is loaded. I can't explain the reason why it happens. I suspect some js magic is there.